Automattic/wp-calypso
# A4A: Add Initial payment methods section.

This pull request introduces the Payment Methods section to the A4A Portal. The pages only have the headers for now as the initial content.


<img width="1416" alt="Screenshot 2024-03-04 at 7 32 20 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/22d7a2ff-03e8-46ee-a0aa-91c4ae5797b9">
<img width="1417" alt="Screenshot 2024-03-04 at 7 32 30 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/50c3c10e-1cf6-4ea8-863a-2866613b4031">



Closes https://github.com/Automattic/jetpack-genesis/issues/273
Closes https://github.com/Automattic/jetpack-genesis/issues/274
Closes https://github.com/Automattic/jetpack-genesis/issues/276

## Proposed Changes

* Add new **'/purchases/payment-methods'** and **'/purchases/payment-methods/add'** paths in the `a8c-for-agencies-purchases` section.
* Add initial page context with correct page headers based from original design (Jetpack Manage).
* Add new 'Purchases > Payment Methods' menu item in the Purchases section sidebar.

## Testing Instructions

* Switch branch: `git checkout add/a4a-payment-method-section`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/purchases/payment-methods.
* Confirm that the Payment Methods overview page is visible.
* Confirm that the 'Purchases > Payment Methods' menu item is visible in the sidebar.
* Click the 'Add new card' button in the page header.
* Confirm that the page redirects to the Payment Methods Add page.


## Pre-merge Checklist
